### PR TITLE
Add hide_none option for the AR header.

### PR DESCRIPTION
**Pull request by dev-aaront-org**
I'm using separate instances of authentication_milter on my inbound servers (doing checks) and my outbound servers (doing signing/sealing), and this option is useful to avoid adding an unnecessary AR header on the outbound side.

Note that these tests only pass with PRs #70 and #72 applied. The tests cover most of the handler hide_none options, as well as the one being added.

I also wanted to say thank you for this project. It is exactly what I was looking for, and it's very easy to add my own handlers to extend it. Thanks!

### CODE REVIEW

**Comment by marcbradshaw** [comment]
Hi,
Thanks for the comments, and for all the Pull Requests.
I'm trying to understand your use case here, is this...

Mail received by inbound, and is checked, but ARC is not checked added (is arc checked?). AR headers are added.
Something internal happens, does this modify the mail?
Mail received by outbound, is sealed, ARC header set is added based on the AR header added by inbound?

**Comment by dev-aaront-org** [comment]
Yes, you got it. I'm using the authentication_milter on the inbound side to check SPF, DKIM, ARC, and DMARC, then pretty typical spam/virus/etc. checking and mailing list expansion could modify the message, then the authentication_milter on the outbound side does DKIM signing and ARC sealing. The ARC sealing does use the ARC result from the AR header added on the inbound side (using Mail::DKIM::ARC::Signer with Chain => ar).

I ended up using this option in a second case as well. I decided to use authentication_milter to DKIM sign mail originating from my org, and since it's mail coming from my users/systems, there really aren't any useful checks to do on it. So I set hide_none to skip adding the AR header, and my DKIM signing handler just adds the DKIM-Signature header(s).

File: bin/authentication_milter [hunk 1]
```diff
@@ -322,6 +322,8 @@ __END__
     "dns_servfail_timeout"  : 1000000,                  | How long (microseconds) a SERVFAIL can take before being considered a timeout
                                                         |     by the internal resolver
 
+    "hide_none"             : 0,                        | Do not add the Authentication-Results header if the result is 'none'
```
Comment (marcbradshaw) [inline bin/authentication_milter:325]: hide_none feels like the wrong name for this, we're not hiding any 'none' results (like with the similarly named DMARC option), we are hiding the entire AR header.

Comment (marcbradshaw) [reply bin/authentication_milter:325]: ah, but only if it would be 'none' gotcha

File: lib/Mail/Milter/Authentication/Handler.pm [hunk 2]
```diff
@@ -2517,7 +2517,9 @@ sub add_headers {
         $header .= '; none';
     }
 
-    $self->prepend_header( 'Authentication-Results', $header );
+    if ( @auth_headers || !$config->{'hide_none'} ) {
```
Comment (marcbradshaw) [inline lib/Mail/Milter/Authentication/Handler.pm:2520]: The flow here feels a little clumsy, I would move the prepend_header up into the if @auth_headers block, add an 'elsif (!$config->{hide_none}) {' block which adds the none result, and a final else block which is a nop with a comment.

Comment (dev-aaront-org) [reply lib/Mail/Milter/Authentication/Handler.pm]: SGTM, I've updated the patch.

**Pull request merged**